src/librustc_lint/builtin.rs

```diff
@@ -738,21 +738,22 @@ impl EarlyLintPass for DeprecatedAttr {
     }
 }
 
-declare_lint! {
-    pub UNUSED_DOC_COMMENTS,
-    Warn,
-    "detects doc comments that aren't used by rustdoc"
+trait UnusedDocCommentExt {
+    fn warn_if_doc(
```

Comment:
The trait is unnecessary, I only used it to minimize the diff.
This can be a free function.